### PR TITLE
Document Cypress 10

### PR DESCRIPTION
**Pull request by alexplischke**
### Description
Document Cypress 10.

### CODE REVIEW

File: docs/web-apps/automated-testing/cypress/yaml-configuration/yaml-v1.md [hunk 1]
```diff
@@ -0,0 +1,893 @@
...
+```
+
+:::note YAML Required
+While you can use multiple files of different names or locations to specify your configurations, each file must be a `*.yml` and follow the `saucectl` syntax. Our IDE Integrations (e.g., [Visual Studio Code](/dev/cli/saucectl/usage/ide/vscode)) can help you out by validating the YAML files and provide handy suggestions, so make sure to check them out!
+:::
+
+
+## Example Configuration
+
+<!-- TODO need to create a new Cypress 10 example first. -->
```
Comment (alexplischke) [inline docs/web-apps/automated-testing/cypress/yaml-configuration/yaml-v1.md:34]: Reminder to self before merging.

**Comment by wswebcreation** [comment]
@alexplischke 

I was just checking this and I understand that we can't mentioned every change from major Cypress versions here, but in V10 they deprecated the `config.json`, see [this](https://docs.cypress.io/guides/references/legacy-configuration#cypress-json) and replaced it with `cypress.config.js`, see [this](https://docs.cypress.io/guides/references/configuration).
It might be useful to also note that in our docs. What do you think?

**Comment by alexplischke** [comment]
> @alexplischke
> 
> I was just checking this and I understand that we can't mentioned every change from major Cypress versions here, but in V10 they deprecated the `config.json`, see [this](https://docs.cypress.io/guides/references/legacy-configuration#cypress-json) and replaced it with `cypress.config.js`, see [this](https://docs.cypress.io/guides/references/configuration). It might be useful to also note that in our docs. What do you think?

It's just one of many changes though. The user absolutely _has_ to go through the [migration process](https://docs.cypress.io/guides/references/migration-guide#Migrating-to-Cypress-version-10-0) that Cypress prescribes for Cypress 10. I would much rather put up a generic note. So let me do that at least 👍

File: docs/web-apps/automated-testing/cypress/yaml/v1.md [hunk 2]
```diff
@@ -0,0 +1,893 @@
...
+:::caution
+Since environment variables are provided to Cypress directly, avoid using `CYPRESS_` as a prefix.
+:::
+
+---
+
+#### `specPattern`
+<p><small>| REQUIRED | STRING/ARRAY/REGEX |</small></p>
+
+One or more paths to the Cypress test files to run for this suite. Regex values are supported to indicate all files of a certain type or in a certain directory, etc.
```
Comment (tianfeng92) [inline docs/web-apps/automated-testing/cypress/yaml/v1.md:803]: It should be glob pattern for `sepcPattern` and `excludeSpecPattern`. https://docs.cypress.io/guides/references/configuration#e2e

File: docs/web-apps/automated-testing/cypress/yaml/v1alpha.md [hunk 3]
```diff
@@ -0,0 +1,898 @@
...
+:::caution
+Since environment variables are provided to Cypress directly, avoid using `CYPRESS_` as a prefix.
+:::
+
+---
+
+#### `testFiles`
+<p><small>| REQUIRED | STRING/ARRAY/REGEX |</small></p>
+
+One or more paths to the Cypress test files to run for this suite, if not otherwise specified explicitly in `cypress.json`. Regex values are supported to indicate all files of a certain type or in a certain directory, etc.
```
Comment (tianfeng92) [inline docs/web-apps/automated-testing/cypress/yaml/v1alpha.md:800]: hmm.. also should be glob pattern for `testFiles` and `excludedTestFiles` https://docs.cypress.io/guides/references/legacy-configuration#Folders-Files

**Review by tianfeng92 — CHANGES_REQUESTED** [review]

**Review by tianfeng92 — APPROVED** [review]

**Pull request merged**